tuna/connection.py

```diff
@@ -177,43 +180,50 @@ def exec_command_unparsed(self, cmd, timeout=SSH_TIMEOUT, abort=None):
     return None, None, None
 
   def exec_command(self, cmd, timeout=SSH_TIMEOUT, abort=None, proc_line=None):
+    # pylint: disable=too-many-nested-blocks, too-many-branches
     """Function to exec commands"""
     _, o_var, e_var = self.exec_command_unparsed(cmd, timeout, abort)
-
-    if not o_var:
-      return 1, o_var, e_var
-
-    if not proc_line:
-      proc_line = lambda x: self.logger.info(line.strip())
-    ret_out = StringIO()
-    ret_err = e_var
     try:
-      while True:
-        line = o_var.readline()
-        if line == '' and not self.is_alive():
-          break
-        if line:
-          proc_line(line)
-          ret_out.write(line)
-      ret_out.seek(0)
-      if self.local_machine:
-        ret_code = self.subp.returncode
-        if ret_out:
-          ret_out.seek(0)
-          ret_err = StringIO()
-          err_str = ret_out.readlines()
-          for line in err_str[-5:]:
-            ret_err.write(line)
-          ret_err.seek(0)
-          ret_out.seek(0)
-      else:
-        ret_code = self.out_channel.recv_exit_status()
-    except (socket.timeout, socket.error) as exc:
-      self.logger.warning('Exception occurred %s', exc)
-      ret_code = 1
-      ret_out = None
-
-    return ret_code, ret_out, ret_err
+      if not o_var:
+        return 1, o_var, e_var
+
+      if not proc_line:
+        # pylint: disable-next=unnecessary-lambda-assignment ; more readable
```

Comment:
thought u fixed this already?

Reply:
thought it was a little more readable, so just disabled the warning